### DISCUSSION TITLE
MDX and <Link>

### DISCUSSION THREAD

**Discussion by Southclaws** [vercel/next.js#11110] [Help]
It seems next-mdx does not turn `[]()` into <Link> to facilitate client-side transitions.

It also seems to be causing a bug where any link that's clicked loads the `/` page, even though the address bar contains the correct desired route.

Is this a misconfiguration, or a bug, or is there something I'm missing?

Using latest 9.3, reduced everything to a simple index page and the issue persists.

**Comment by laurentsenta** [comment]
Hi @Southclaws 

have you tried configuring the mdx components?

Here is my `_app.tsx`, links work:

```tsx
import React from 'react'
// @ts-ignore
import {MDXProvider} from '@mdx-js/react'
import Link from 'next/link'
import { AppProps } from 'next/app'

const components = {
  a: Link
  /*
  h1: Heading.H1,
  h2: Heading.H2,
  // …
  p: Text,
  code: Pre,
  inlineCode: Code
  */
}

export default function App(app: AppProps) {
  const { Component, pageProps } = app;
  return (<MDXProvider components={components}>
      <Component {...pageProps} />
    </MDXProvider>
  );
}
```

I hope that's helpful.

**Reply by leerob** [reply]
Building off this, consider checking if it's an internal route for client-side transitions. Something kind of [like this](https://github.com/leerob/leerob.io/blob/master/components/MDXComponents.js#L52-L71):

```js
const isInternalLink = href && (href.startsWith('/') || href.startsWith('#'));

if (isInternalLink) {
    return (
        <Link href={href}>
            <a {...props} />
        </Link>
    );
}

return <a />;
```

**Reply by AnweshGangula** [reply]
for anyone looking for this - this is the [updated link](https://github.com/leerob/leerob.io/blob/main/app/components/mdx.tsx#L34) to leerob's code referenced above. Also check out this related post: https://github.com/vercel/next.js/issues/50183#issuecomment-1562626044
